gnosis/safe/safe_signature.py

```diff
@@ -50,6 +51,19 @@ def from_v(v: int):
             return SafeSignatureType.EOA
 
 
+def uint_to_address(value: int) -> ChecksumAddress:
+    """
+    Convert a Solidity `uint` value to a checksummed `address`, removing
+    invalid padding bytes if present
+
+    :return: Checksummed address
+    """
+    encoded = encode_single("uint", value)
+    # Remove padding bytes, as Solidity will ignore it but `eth_abi` will not
+    encoded_without_padding_bytes = b"\x00" * 12 + encoded[-20:]
```

Comment:
I would have expected the python lib to handle this.

Reply:
Yeah, I would too 😢 : https://github.com/ethereum/eth-abi/issues/162